src/app/handlers/command.handler.ts

```diff
@@ -1,8 +1,24 @@
 import * as types from '../../common/types';
 import Constants from '../../common/constants';
 import levenshtein from 'js-levenshtein';
-import { MessageEmbed, MessageReaction, User } from 'discord.js';
+import { CommandInteraction, MessageEmbed, MessageReaction, User } from 'discord.js';
 import ms from 'ms';
+import ISlashPlugin from '../../common/slash';
+
+type CommandResolvable =
+  | {
+      type: 'message';
+      message: types.IMessage;
+      plugin: types.IPlugin;
+      command: types.ICommand;
+      isDM: boolean;
+    }
+  | {
+      type: 'interaction';
+      message: CommandInteraction;
```

Comment:
do we want to have a `types.ICommandInteraction`?

Reply:
What would be the idea here? Just a wrapper around the `CommandInteraction` type? If so, seems like something we can add later if we need it.